src/inmanta/execute/runtime.py

```diff
@@ -459,27 +470,42 @@ def __init__(self, attribute: "Attribute", instance: "Instance", queue: "QueueSc
         self.myself = instance
         self.location = None
 
-    def set_value(self, value: "Instance", location: Location, recur: bool = True) -> None:
+    def _get_null_value(self) -> object:
+        return None
```

Comment:
Why does this one return None instead of NoneValue? (I am not very familiar with the compiler code)

Reply:
I'd say mostly legacy reasons. `NoneValue` is used for nullable attributes, while `None` is used here to indicate the absence of any values. @wouterdb might be able to give more context.
Now that you point it out it is a bit confusing that we allow assigning `NoneValue` but we don't actually use it in the internal representation. We're treating one value (`null`) different for two different types of attributes, which is not very consistent. I'm not sure about a better approach at the moment though, I'd have to think about it.